modules/margin-liquidity-pools/src/lib.rs

```diff
@@ -156,12 +165,28 @@ decl_module! {
 			Self::deposit_event(RawEvent::SetEnabledTrades(who, pool_id, pair, enabled));
 		}
 
-		pub fn update_swap(origin, pool_id: LiquidityPoolId, pair: TradingPair, rate: Fixed128) {
+		pub fn set_swap_rate(origin, pair: TradingPair, rate: SwapRate) {
+			T::UpdateOrigin::try_origin(origin)
+				.map(|_| ())
+				.or_else(ensure_root)?;
+
+			let max_swap = T::MaxSwap::get();
+			ensure!(rate.long >= max_swap.long, Error::<T>::SwapRateTooLow);
+			ensure!(rate.short <= max_swap.short, Error::<T>::SwapRateTooHigh);
```

Comment:
Both long or short rate could be either positive or negative, compare against their abs.